tests/pytests/test_resp3.py

```diff
@@ -708,3 +708,374 @@ def test_profile_child_itrerators_array():
+    env.expect('FT.AGGREGATE', 'idx2', '*', 'FORMAT', 'EXPAND').equal(
+       {'attributes': [], 'error': [], 'total_results': 0, 'format': 'EXPAND', 'results': []}
+    )
+
+
+def testExpandErrorsResp2():
+  env = Env(protocol=2)
+  env.cmd('ft.create', 'idx', 'on', 'json', 'SCHEMA', '$.arr', 'as', 'arr', 'numeric')
+  env.expect('FT.SEARCH', 'idx', '*', 'FORMAT', 'EXPAND').error()
+  if not env.isCluster():
+    env.expect('FT.AGGREGATE', 'idx', '*', 'FORMAT', 'EXPAND').error()
+  else:
+    # TODO: Expect an error once MOD-5211 is done
+    env.expect('FT.AGGREGATE', 'idx', '*', 'FORMAT', 'EXPAND').equal([0])
+
+  # On HASH
+  env.cmd('ft.create', 'idx2', 'on', 'hash', 'SCHEMA', 'num', 'numeric', 'str', 'text')
+  if not env.isCluster():
+    env.expect('FT.AGGREGATE', 'idx2', '*', 'FORMAT', 'EXPAND').error()
+  else:
+    # TODO: Expect an error once MOD-5211 is done
+    env.expect('FT.AGGREGATE', 'idx2', '*', 'FORMAT', 'EXPAND').equal([0])
+
+def testExpandJson():
+  ''' Test returning values for JSON in expanded format (raw RESP3 instead of stringified JSON) '''
+  env = Env(protocol=3)
+  env.cmd('ft.create', 'idx', 'on', 'json', 'SCHEMA',
+          '$.arr', 'as', 'arr','numeric',
+          '$.num', 'as', 'num', 'numeric',
+          '$.str', 'as', 'str', 'text',
+          '$..arr[*]', 'as', 'multi', 'numeric')
+
+  with env.getClusterConnectionIfNeeded() as r:
+    r.execute_command('json.set', 'doc1', '$', '{"arr":[1.0,2.1,3.14],"num":1,"str":"foo","sub":{"s1":false},"sub2":{"arr":[10,20,33.33]}, "empty_arr":[], "empty_obj":{}}')
+    r.execute_command('json.set', 'doc2', '$', '{"arr":[3,4,null],"num":2,"str":"bar","sub":{"s2":true},"sub2":{"arr":[40,50,66.66]}, "empty_arr":[], "empty_obj":{}}')
+    r.execute_command('json.set', 'doc3', '$', '{"arr":[5,6,7],"num":3,"str":"baz","sub":{"s3":false},"sub2":{"arr":[70,80,99.99]}, "empty_arr":[], "empty_obj":{}}')
+
+  exp_string = {
+    'attributes': [],
+    'error': [],
+    'total_results': ANY,
+    'format': 'STRING',
+    'results': [
+      {'id': 'doc1', 'extra_attributes': {'num': '1', '$': '{"arr":[1.0,2.1,3.14],"num":1,"str":"foo","sub":{"s1":false},"sub2":{"arr":[10,20,33.33]},"empty_arr":[],"empty_obj":{}}'}, 'values': []},
+      {'id': 'doc2', 'extra_attributes': {'num': '2','$': '{"arr":[3,4,null],"num":2,"str":"bar","sub":{"s2":true},"sub2":{"arr":[40,50,66.66]},"empty_arr":[],"empty_obj":{}}'}, 'values': []},
+    ]
+  }
+  exp_expand = {
+    'attributes': [],
+    'error': [],
+    'total_results': ANY,
+    'format': 'EXPAND',
+    'results': [
+      {'id': 'doc1', 'extra_attributes': {'num': [1], '$': [{"arr": [1, 2.1, 3.14], "num": 1, "str": "foo", "sub":{"s1": 0}, "sub2":{"arr": [10, 20, 33.33]}, "empty_arr":[],"empty_obj":{}}]}, 'values': []},
+      {'id': 'doc2', 'extra_attributes': {'num': [2], '$': [{"arr": [3, 4, None], "num": 2, "str": "bar", "sub":{"s2": 1 }, "sub2":{"arr": [40, 50, 66.66]}, "empty_arr":[],"empty_obj":{}}]}, 'values': []},
+    ]
+  }
```

Comment:
please use `json.dumps` over the expected object, so once you chane the test it'll be one change. Also you'll be more pro ;)